### PR TITLE
Add E2E tests shopper checkout

### LINKED ISSUE #2485: E2E tests: Shopper - Checkout
- [x] [Shopper - Checkout: Checkout with normal credit card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-normal-credit-card)
- [x] [Shopper - Checkout: Checkout with saved card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-saved-card)
- [x] [Shopper - Checkout: Checkout with SCA card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-sca-card)
- [x] [Shopper - Checkout: Checkout failures (with one of the error cards)](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-failures-with-various-cards)
- [ ] [Shopper - Checkout: Checkout with Payment Request](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-payment-request)
- [ ] [Shopper - Checkout: Checkout with Payment Request (3DS card)](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-3ds-card)

### PR DESCRIPTION
**Pull request by rafaelzaleski**
Closes #2485 

## Changes proposed in this Pull Request:

This PR adds the following E2E test cases:

[Shopper - Checkout: Checkout with normal credit card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-normal-credit-card)
 [Shopper - Checkout: Checkout with saved card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-saved-card)
 [Shopper - Checkout: Checkout with SCA card](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-with-sca-card)
 [Shopper - Checkout: Checkout failures](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Testing-instructions-for-critical-flows-using-the-classic-checkout-experience#checkout-failures-with-various-cards)

## Testing instructions

1. Follow the instructions [here](https://github.com/woocommerce/woocommerce-gateway-stripe/tree/develop/tests/e2e)  to setup a new E2E environment using a jurassic ninja site.
2. Run the E2E tests.

All tests should pass.

---

-   [x] Covered with tests (or have a good reason not to test in description ☝️)
-   [x] Added changelog entry **in both** `changelog.txt` and `readme.txt` (or does not apply)
-   [x] Tested on mobile (or does not apply)

**Post merge**

-   [ ] Added testing instructions to the [Release Testing Instructions wiki page](https://github.com/woocommerce/woocommerce-gateway-stripe/wiki/Release-Testing-Instructions) (or does not apply)

### CODE REVIEW

File: tests/e2e/tests/checkout/card-failures.spec.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+const { test, expect } = require( '@playwright/test' );
```
Comment (ricardo) [inline tests/e2e/tests/checkout/card-failures.spec.js]: It looks like we're not using `require` throughout the repo - except for webpack.config and the release script.

Can we use only `import` over `require`, for consistency, or is there a reason to use `require` in some places?

File: tests/e2e/tests/checkout/card-failures.spec.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+	fillCardDetails,
+} = require( '../../utils/payments' );
+
+test.beforeEach( async ( { page } ) => {
+	await emptyCart( page );
+	await setupProductCheckout( page );
+	await setupCheckout( page, config.get( 'addresses.customer.billing' ) );
+} );
+
+const testFunctionForCard = async ( page, cardKey ) => {
```
Comment (ricardo) [inline tests/e2e/tests/checkout/card-failures.spec.js]: Nit: Does the "Function" in the function name stand for "functionality", or is it a redundancy thing, like naming a variable A with `variableA` ? 😀 

In both cases, I'd suggest something like `testCardFailure`, but feel free to ignore.

File: tests/e2e/tests/checkout/saved-card.spec.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+		await page.waitForNavigation();
+		await expect( page.locator( 'h1.entry-title' ) ).toHaveText(
+			'Order received'
+		);
+	} );
+
+	await test.step( 'checkout and pay with the saved card', async () => {
+		await emptyCart( page );
+		await setupProductCheckout( page );
+		await setupCheckout( page, null, true );
```
Comment (ricardo) [inline tests/e2e/tests/checkout/saved-card.spec.js:62]: It looks like these 3 steps are being used, and are gonna be needed for many future test cases.

I think it's fine as is, but we might consider moving it to a single function later.

**Review by ricardo — COMMENTED** [review]
Tests well, nice work!

I found it great that you were able to add so many test cases in a clean way and with a relatively small amount of code.

I left a comment regarding the use of `require`, but other than that, this LGTM.

File: tests/e2e/tests/checkout/card-failures.spec.js (hunk 1 above)
Comment (rafaelzaleski) [reply tests/e2e/tests/checkout/card-failures.spec.js]: It's just that I started re-using the code from [#woocommerce/](https://github.com/woocommerce/woocommerce) and they were using `require` on their codebase. I've replaced all `require` with `import`.

File: tests/e2e/tests/checkout/card-failures.spec.js (hunk 2 above)
Comment (rafaelzaleski) [reply tests/e2e/tests/checkout/card-failures.spec.js]: Good catch! It was redundancy like using `variableA` for a variable :laughing: 
The name was changed to `async testCard( page, cardKey )`.

File: tests/e2e/tests/checkout/saved-card.spec.js (hunk 3 above)
Comment (rafaelzaleski) [reply tests/e2e/tests/checkout/saved-card.spec.js:62]: I'll keep that in mind for the next PR with test cases.

File: tests/e2e/utils/pw-setup.js [hunk 4]
```diff
@@ -1,10 +1,14 @@
-import { resolve } from 'path';
+require( 'dotenv' ).config( {
```
Comment (ricardo) [inline tests/e2e/utils/pw-setup.js:1]: Here and in `global-setup`, we couild use `import` too:

```javascript
import * as dotenv from 'dotenv';
dotenv.config( {
    path: `${ process.env.E2E_ROOT }/config/local.env`,
} );
```

**Review by ricardo — APPROVED** [review]
LGTM and tests well. :shipit: 

Just left one more comment regarding the use of `import` over `replace`. Feel free to merge after looking into that.

File: tests/e2e/utils/pw-setup.js (hunk 4 above)
Comment (rafaelzaleski) [reply tests/e2e/utils/pw-setup.js:1]: nice! I'll update in [this PR](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/2517)

**Pull request merged**